### ISSUE TITLE
User Settings screen allows saving of invalid settings, logging user out, not able to log back in

### ISSUE THREAD

**Issue by kopancek** [sourcegraph/sourcegraph#42000]
User Settings screen allows saving of invalid settings, logging user out, not able to log back in

#### Steps to reproduce:

https://www.loom.com/share/94b37a539e114e5a9dc55706fea367bf

1. Go to Settings -> User Settings
2. Comment out everything
3. Save

The user is logged out. Any try to log back in is going to fail

#### Expected behavior:

1. Saving of invalid JSON (that is either completely invalid or does not follow the schema) is not possible.
2. Making the request to save invalid config via graphql should result in an error (it's not enough to prevent this behavior on the UI)

#### Actual behavior:

What you see in the loom video.

/cc @sourcegraph/iam

**Comment by leonore** [comment]
so this is not lost in slack, just in case this isn't fixed soon and users run into this, this is how we fixed it:
```sql
select * from settings where user_id = (select id from users where username = 'username’) order by id desc ;
update settings set contents = (select contents from settings where id =2632) where id =2666;
```
the numerical idea corresponds to the id of the previous unbroken settings version that was stored

**Comment by sqs** [comment]
Agree that you should not be able to save syntactically invalid JSON.

For JSON that is invalid with respect to the JSON Schema, that requires a bit more thought. The JSON Schema changes, and we currently don't always migrate settings along with the JSON Schema. If we enforce schema validity, then we'll need full forward and backward migrations for JSON settings. This could be a good idea, but it is more work and should probably be done separately from the syntactic validation.

**Comment by sqs** [comment]
It looks like we do catch syntax errors in the JSON, which is good. But things break if the JSON is just comments (which is syntactically correct, but still weird).

**Comment by sqs** [comment]
Fix in https://github.com/sourcegraph/sourcegraph/pull/42035

**Comment by kopancek** [comment]
Closed the issue as @sqs already fixed the main problem for us 🙇 in #42035
